concourse/scripts/s3init.sh

```diff
@@ -4,12 +4,29 @@ bucket=$1
 file=$2
 init_file=$3
 
+# Attempt to use instance profile if keys not configured
+if [[ -z "${AWS_SECRET_ACCESS_KEY}" ]] && [[ -z ${AWS_ACCESS_KEY_ID} ]] ; then
+  meta_url="http://169.254.169.254/latest/meta-data/iam/security-credentials/"
+  profile_name=$(curl -s ${meta_url})
+  instance_profile=$(curl -s ${meta_url}/${profile_name})
+
+      AWS_ACCESS_KEY_ID=$(echo "${instance_profile}" | grep AccessKeyId     | cut -d':' -f2 | sed 's/[^0-9A-Z]*//g')
+     AWS_SECURITY_TOKEN=$(echo "${instance_profile}" | grep Token           | cut -d':' -f2 | sed 's/[^0-9A-Za-z/+=]*//g')
+  AWS_SECRET_ACCESS_KEY=$(echo "${instance_profile}" | grep SecretAccessKey | cut -d':' -f2 | sed 's/[^0-9A-Za-z/+=]*//g')
```

Comment:
The following might be a little safer and simpler, because it's not using any regexes:
```
AWS_ACCESS_KEY_ID=$(echo "${instance_profile}" | awk -F\" '$2 == "AccessKeyId" { print $4 }'
AWS_SECURITY_TOKEN=$(echo "${instance_profile}" | awk -F\" '$2 == "Token" { print $4 }'
AWS_SECRET_ACCESS_KEY=$(echo "${instance_profile}" | awk -F\" '$2 == "SecretAccessKey" { print $4 }'
```
Although we might need to double check if any of these tools are still available if the container changes in alphagov/paas-docker-cloudfoundry-tools#7